BaristaSwift/Common/Common/Store/LocalOrderStore.swift

```diff
@@ -73,31 +74,19 @@ public class LocalOrderStore {
         return orders
     }
     
-    public func completeOrder(_ order:LocalOrder, completion:@escaping (Bool) -> Void) {
+    public func completeOrder(_ order:LocalOrder) -> Promise<Void> {
         let opty = order.opportunity
         opty.stage = .closedWon
         let quote = order.quote
         quote.status = .accepted
         
-        QuoteStore.instance.updateEntry(entry: quote) { (quoteSync) in
-            guard let quoteState = quoteSync else { return }
-            if quoteState.isDone() {
-                SalesforceSwiftLogger.log(type(of:self), level:.info, message:"update quote completed")
-                OpportunityStore.instance.updateEntry(entry: opty, completion: { (optSync) in
-                    guard let optyState = optSync else { return }
-                    if optyState.isDone() {
-                        SalesforceSwiftLogger.log(type(of:self), level:.info, message:"update opty completed")
-                        completion(true)
-                    } else if optyState.hasFailed() {
-                        SalesforceSwiftLogger.log(type(of:self), level:.info, message:"Failed syncing opportunity update")
-                        completion(false)
-                    }
-                })
-            } else if quoteState.hasFailed() {
-                SalesforceSwiftLogger.log(type(of:self), level:.info, message:"Failed syncing quote update")
-                completion(false)
+        return QuoteStore.instance.updateEntry(entry: quote)
```

Comment:
Notice how much shorter the code is because I don't have to deal with the sync state and also I get the updated entry back from updateEntry() (and it's guaranteed to be an object).